### PR TITLE
Unit 2 assess jamar

### PR DESCRIPTION
**Pull request by Jwat5623**
# Submitting Unit 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [X ] Coding Challenges
- [X] Coding Concepts
- [X ] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,39 +14,103 @@
...
+        // Expected output: "15 is divisible by three"
+        const object2 = { number: 0 }
+        // Expected output: "0 is divisible by three"
+        const object3 = { number: -7 }
+        // Expected output: "-7 is not divisible by three"
+        expect(divByThree(15)).toEqual("15 is divisible by three") 
+        expect(divByThree(0)).toEqual("0 is divisible by three")
+        expect(divByThree(-7)).toEqual("-7 is not divisible by three")
+    })
+})
```
Comment (sjproctor) [inline code-challenges.test.js:57]: Your test structure looks great, but you aren't actually using your test variables so you are coding for the wrong input.

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,39 +14,103 @@
...
+const allCaps = (array) => {
+    return array.map((value) => {
+      if (typeof value === 'string' || value instanceof String) {
+        return value.charAt(0).toUpperCase() + value.slice(1)
+      } else {
+        return value
+      }
+    })
+}
```
Comment (sjproctor) [inline code-challenges.test.js:103]: Interesting edge case you are solving for with your typeof and instanceof.

File: coding-concepts.js [hunk 3]
```diff
@@ -9,28 +9,28 @@
...
 const greeter = (name) => {
   `Hello, ${name}!`
 }
-// console.log(greeter("LEARN Student"))
+console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: it will be an error
+// b) Verify and explain: I was close, it printed undefined because not enough information was put into the function
```
Comment (sjproctor) [inline coding-concepts.js:25]: Specifically, the keyword return was missing.

File: coding-concepts.js [hunk 4]
```diff
@@ -55,7 +55,7 @@ class Learn {
   }
 }
 const learnStudent = new Learn("George")
-// console.log(learnStudent)
+console.log(learnStudent)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: it will add George as the student name
+// b) Verify and explain: George was added as a value inside of the student key
```
Comment (sjproctor) [inline coding-concepts.js:61]: `learnStudent` is an object that is created from the class.

**Review by sjproctor — APPROVED** [review]
Nice work on your interview answers! Your code is showing an advanced beginner level of skill. Focus on the structure of the test and make sure you are using your test variables as intended!
